unixfs/io/dirbuilder.go

```diff
@@ -29,22 +27,13 @@ func NewDirectory(dserv mdag.DAGService) *directoryBuilder {
 }
 
 // AddChild adds a (name, key)-pair to the root node.
-func (d *directoryBuilder) AddChild(name string, k key.Key) error {
-	// TODO(cryptix): consolidate context managment
-	ctx, cancel := context.WithTimeout(context.TODO(), time.Minute)
```

Comment:
Ideally, most of the places we do this (context.TODO with a timeout) should be replaced with a passed in context. and the caller can decide the timeout if they want.